public/pages/Indices/containers/Indices/Indices.tsx

```diff
@@ -0,0 +1,206 @@
+import { DEFAULT_PAGE_SIZE_OPTIONS, DEFAULT_QUERY_PARAMS, indicesColumns } from "../../utils/constants";
+import { ModalConsumer } from "../../../../components/Modal";
+import IndexService from "../../../../services/IndexService";
+import { TableParams } from "../../../../models/interfaces";
+import { CatIndex } from "../../../../../server/models/interfaces";
+import { getURLQueryParams } from "../../utils/helpers";
+import { PoliciesQueryParams } from "../../models/interfaces";
+import { BREADCRUMBS } from "../../../../utils/constants";
+
+interface IndicesProps extends RouteComponentProps {
+  indexService: IndexService;
+}
+
+interface IndicesState {
+  totalIndices: number;
+  from: number;
+  size: number;
+  search: string;
+  sortField: string;
+  sortDirection: string;
+  selectedItems: CatIndex[];
+  indices: CatIndex[];
+  loadingIndices: boolean;
+}
+
+export default class Indices extends Component<IndicesProps, IndicesState> {
+  constructor(props: IndicesProps) {
+    super(props);
+
+    const { from, size, search, sortField, sortDirection } = getURLQueryParams(this.props.location);
+
+    this.state = {
+      totalIndices: 0,
+      from,
+      size,
+      search,
+      sortField,
+      sortDirection,
+      selectedItems: [],
+      indices: [],
+      loadingIndices: true,
+    };
+
+    this.getIndices = _.debounce(this.getIndices, 500, { leading: true });
+  }
+
+  async componentDidMount() {
+    chrome.breadcrumbs.set([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDICES]);
+    await this.getIndices();
+  }
+
+  async componentDidUpdate(prevProps: IndicesProps, prevState: IndicesState) {
+    const prevQuery = Indices.getQueryObjectFromState(prevState);
+    const currQuery = Indices.getQueryObjectFromState(this.state);
+    if (!_.isEqual(prevQuery, currQuery)) {
+      await this.getIndices();
+    }
+  }
+
+  static getQueryObjectFromState({ from, size, search, sortField, sortDirection }: IndicesState): PoliciesQueryParams {
+    return { from, size, search, sortField, sortDirection };
+  }
+
+  getIndices = async (): Promise<void> => {
+    this.setState({ loadingIndices: true });
+    try {
+      const { indexService, history } = this.props;
+      const { from, size, search, sortField, sortDirection } = this.state;
+      const params = { from, size, search, sortField, sortDirection };
+      const queryParamsString = queryString.stringify(params);
+      history.replace({ ...this.props.location, search: queryParamsString });
+      const getIndicesResponse = await indexService.getIndices(queryParamsString);
+      if (getIndicesResponse.ok) {
+        const { indices, totalIndices } = getIndicesResponse.response;
+        this.setState({ indices, totalIndices });
+      } else {
+        toastNotifications.addDanger(getIndicesResponse.error);
+      }
+    } catch (err) {
+      toastNotifications.addDanger(err.message || err.data.message || "There was a problem LOADING the indices, please try again.");
```

Comment:
It's some AngularHttpError:
```
export function isAngularHttpError(error: any): error is AngularHttpError {
  return (
    error &&
    typeof error.status === 'number' &&
    typeof error.statusText === 'string' &&
    error.data &&
    typeof error.data.message === 'string'
  );
}
```

This could break if it's not an AngularHttpError and err.message is not filled out. Will add a helper method to constructor error message instead to use around app.